documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_moderation_metrics.md

```diff
@@ -0,0 +1,334 @@
+parallel tracks of work, one for the real time metrics and one for the deferred
+metrics.
+
+## Step details
+
+<!--
+Describe all of the implementation steps listed in the "step-by-step plan" in detail.
+
+For each step description, ensure the heading includes an obvious reference to the step as described in the
+"step-by-step plan" section above.
+-->
+
+```{caution}
+This document shows code-samples but intended as a guide and not meant to be
+used directly!
+```
+
+### Identify and categorise metrics
+
+The metrics we aim to measure can be classified along two axes:
+
+- Whether they measure user-submitted reports or moderator-generated responses
+- Whether they are real-time or deferred
+
+The following metrics can be considered to begin with (we can always add more in
+the future):
+
+- Metrics about reports
+  - number of reports (time-series, real-time)
+    <!-- TODO: - per media type (time-series, real-time) -->
+    - per violation (copyright, sensitive, other)
+  - most reported creator/source/provider (deferred)
+  - accuracy of reports (deferred)
+  - duplication in reports (deferred)
+- Metrics about responses
+  - number of decisions (time-series, real-time)
+    <!-- TODO: - per media type (time-series, real-time) -->
+  - number of resolutions (time-series, real-time)
+    <!-- TODO: - per media type (time-series, real-time) -->
+    - per resolution type (confirmation, rejection, deduplication)
+  - time to resolution (deferred)
+    - average
+    - P99
+
+The real-time metrics can go to a visual dashboard where they can be plotted as
+a time-series. They can be used to observe trends and spikes in reports (which
+is useful to understand usage and detect abuse of our reporting mechanisms) and
+in resolutions (which is useful to know how our resolution volume compares to
+reporting volume and when we need to scale our moderation resources).
+
+The deferred metrics are computed periodically from our database. They can be
+used to build reports that can be presented inside Django's admin interface
+(which are useful to identify the sources, providers and creators that have a
+lot of sensitive content and need bulk moderation). Having them in Django Admin
+brings them closer to where the moderation is being performed.
+
+### Implement real-time metrics
+
+Real-time metrics are implemented by emitting structured log lines from the
+moderation pipeline, whenever one of the following event occurs.
+
+- report created
+- report confirmed
+- report rejected
+- report deduplicated
+- decision created
+
+```{todo}
+Find answers to these questions:
+
+- Why does the
+  [Django Admin IP](/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.md)
+  have separate events for `confirmed_sensitive` and `deindexed_sensitive`?
+
+- Additionally shouldn't the action just be `confirmed` (with deindexing being
+  a consequence of that action)?
+```
```

Comment:
I have a few questions that @sarayourfriend can help out with as the author of "Django admin tools and access control for moderators".

Reply:
`confirmed_sensitive` means the moderator has confirmed the user report so we will mark the work sensitive. `deindexed_sensitive` is the same, but includes the additional step of _deindexing_ the work because it was _too_ sensitive (at the discretion of the moderator). `deindexed_sensitive` is our way of indicating a takedown for non-copyright reasons.

> Openverse moderators check these reports and make decisions about whether to add a sensitivity designation to the work or, in certain cases as described above, delist the work from Openverse’s services.
>
> <https://openverse.org/sensitive-content#user-reported-sensitivity>

`confirmed_sensitive` corresponds to the first possible action described there, adding the sensitivity designation. `deindexed_sensitive` corresponds to the second possible action, delisting it from Openverse.

Hopefully that makes sense!